modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPOptionValueLocalServiceImpl.java

```diff
@@ -410,16 +411,33 @@ private int _searchCPOptionValuesCount(SearchContext searchContext)
 		return GetterUtil.getInteger(indexer.searchCount(searchContext));
 	}
 
-	private void _validate(long cpOptionValueId, long cpOptionId, String key)
+	private void _validateKey(long cpOptionValueId, long cpOptionId, String key)
 		throws PortalException {
 
+		if (Validator.isBlank(key)) {
+			throw new CPOptionValueKeyException("Key is null");
+		}
+
+		key = _friendlyURLNormalizer.normalize(key);
```

Comment:
I moved it inside the method to avoid doing a normalize on a possible empty or null string, the alternative would have been to do the normalize before the validate